web/react/components/textbox.jsx

```diff
@@ -27,11 +40,48 @@ module.exports = React.createClass({
     },
     componentWillUnmount: function() {
         PostStore.removeAddMentionListener(this._onChange);
+        ErrorStore.removeChangeListener(this._onError);
     },
     _onChange: function(id, username) {
         if (id !== this.props.id) return;
         this.addMention(username);
     },
+    _onError: function() {
+        var errorState = getStateFromStores();
+
+        if (this.state.timerInterrupt != null) {
+            window.clearInterval(this.state.timerInterrupt);
+            this.setState({ timerInterrupt: null });
+        }
+
+        if (errorState.message === "There appears to be a problem with your internet connection") {
+            this.setState({ connection: " bad-connection" });
+            var timerInterrupt = window.setInterval(this._onTimerInterrupt, 5000);
+            this.setState({ timerInterrupt: timerInterrupt });
+        }
+        else {
+            this.setState({ connection: "" });
+        }
+    },
+    _onTimerInterrupt: function() {
+        //Since these should only happen when you have no connection and slightly briefly after any
+        //performance hit should not matter
+        if (this.state.connection === " bad-connection") {
+            AppDispatcher.handleServerAction({
+                type: ActionTypes.RECIEVED_ERROR,
+                err: null
+            });
+            this.setState({ numPresses: 0 });
+
+            window.clearInterval(this.state.timerInterrupt);
```

Comment:
Also, you could move the clearInterval and setState calls outside of the braces since they occur in both the if and else branches unless the ordering of the call to AsyncClient.updateLastViewedAt is important.